### ISSUE TITLE
`bad_bit_mask` does not handle unsigned bit shift overflow

### ISSUE THREAD

**Issue by arkeet** [rust-lang/rust-clippy#16781]
### Summary

In a comparison of unsigned integers of the form `_ <bit_op> m <cmp_op> c` where `c` is a constant expression with a left shift (such as `x & m == 0x11_u8 << 4`), the lint always fires when the shift overflows (i.e. some bits get shifted out), with the message showing a value that is outside of the type's range. (It appears to work correctly for signed integers.)

Aside: there appears to be no similar lint for a comparison like `x >> 4 == 0xff_u8`, which is also unsatisfiable and is equally problematic.

### Lint Name

bad_bit_mask

### Reproducer

I tried this code:

```rust
fn foo(x: u8) -> bool {
    x & 0xf0 == 0x11 << 4
}
```

I saw this happen:

```
error: incompatible bit mask: `_ & 240` can never be equal to `272`
 --> .../lib.rs:2:5
  |
2 |     x & 0xf0 == 0x11 << 4
```

I expected to see this happen:

No lint, because `0x11 << 4` is `0x10`, which is a feasible value of the LHS. Note that 272 (which is `0x110`) is outside the range of `u8` values.

### Version

```text
rustc 1.94.1 (e408947bf 2026-03-25)
binary: rustc
commit-hash: e408947bfd200af42db322daf0fadfe7e26d3bd1
commit-date: 2026-03-25
host: x86_64-pc-windows-msvc
release: 1.94.1
LLVM version: 21.1.8
```
